openml/extensions/sklearn/extension.py

```diff
@@ -1597,6 +1616,18 @@ def _prediction_to_probabilities(y: np.ndarray, classes: List[Any]) -> np.ndarra
             else:
                 model_classes = used_estimator.classes_
 
+            if not isinstance(model_classes, list):
+                model_classes = model_classes.tolist()
+
+            # to handle the case when dataset is numpy and categories are encoded
+            # however the class labels stored in task are still categories
+            if isinstance(y_train, np.ndarray) and isinstance(
+                typing.cast(List, task.class_labels)[0], str
```

Comment:
This requires `task.class_labels` to be not None. I don't think this is checked at this point?

Reply:
Not directly, but this check is under the scope of an if-statement checking for `OpenMLClassificationTask` and `OpenMLLearningTask` which should imply `task.class_labels` is not `None` for that task.
Moreover, the second half of this if-statement checks if `task.class_labels` is a list of strings, which again is indirectly checking `task.class_labels` to be not `None`.

Do we then need to check for it?
I can still add an explicit check for None and raise an error if you deem necessary.

This though appears to be related to the `Optional` status of `task.class_labels` as you mention in a comment below.